### PR TITLE
use sql for all fluff checkpoints

### PR DESCRIPTION
**Pull request by czue**
(don't merge yet)

### CODE REVIEW

**Comment by snopoke** [comment]
Test failure real:
```
======================================================================
ERROR: test_deleting_on_doc_type_change (tests.tests.Test)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/dimagi/fluff/tests/tests.py", line 524, in test_deleting_on_doc_type_change
    pillow = cls.pillow()(chunk_size=0)
TypeError: __init__() got an unexpected keyword argument 'chunk_size'
======================================================================
ERROR: test_deleting_on_doc_type_change_sql (tests.tests.Test)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/dimagi/fluff/tests/tests.py", line 550, in test_deleting_on_doc_type_change_sql
    pillow = cls.pillow()(chunk_size=0)
TypeError: __init__() got an unexpected keyword argument 'chunk_size'
======================================================================
ERROR: test_indicator_calculation (tests.tests.Test)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/dimagi/fluff/tests/tests.py", line 157, in test_indicator_calculation
    pillow = cls.pillow()(chunk_size=0)
TypeError: __init__() got an unexpected keyword argument 'chunk_size'
```

**Comment by czue** [comment]
thx, look

**Comment by czue** [comment]
:white_check_mark: @snopoke @calellowitz

File: tests/tests.py [hunk 1]
```diff
@@ -556,13 +556,23 @@ def test_deleting_on_doc_type_change_sql(self):
...
                 self.assertEqual(rows.rowcount, 0)
 
+
+def mock_checkpoint():
+    # for some reason the testrunner chokes on these if they are not defined inline.
+    # in the future we may want to explicitly use django tests instead of regular tests
+    # if we're going to depend on a django environment.
+    from pillowtop.checkpoints.manager import PillowCheckpoint
+    from pillowtop.dao.mock import MockDocumentStore
+    return PillowCheckpoint(MockDocumentStore(), 'mock-checkpoint')
```
Comment (czue) [inline tests/tests.py]: not a huge fan of having to mock out the checkpoint in the tests, but the sql checkpoint api is [well-tested in pillowtop](https://github.com/dimagi/pillowtop/blob/master/pillowtop/tests/test_checkpoints.py#L28) and i didn't want to bother with switching the testrunner to bootstrap a whole django environment

**Pull request merged**